### PR TITLE
Ignore while a promise returns inside a test with a value. Only watch for rejection

### PR DESCRIPTION
**Pull request by corbinu**
Addresses #700 by adding a new test for a resolving promise and then updating the runner to ignore resolved values. Let me know if there are any changes you want me to make :)

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -743,7 +743,11 @@ internals.protect = function (item, state, callback) {
         if (itemResult &&
             itemResult.then instanceof Function) {
 
-            itemResult.then(done, done);
+            itemResult.then(() => {
```
Comment (Marsup) [inline lib/runner.js]: Just `itemResult.then(() => done(), done)`.

**Comment by corbinu** [comment]
@Marsup Your suggested change seems to have caused 2 other tests to fail. Thoughts?

**Comment by Marsup** [comment]
Your change doesn't look remotely similar to my suggestion :)

**Comment by Marsup** [comment]
Thanks ! I'll let @geek merge and release as it's not mine to make.

**Pull request merged**